File: 卡码/代码随想录/JS/图论/dfs/LeetCode_0841.js
//841. 钥匙和房间
const canVisitAllRooms = function(rooms) {
  const visited=new Array(rooms.length).fill(false);
  const dfs=(key)=>{
    if(visited[key]) return;
    visited[key]=true;
    const keys=rooms[key];
    for(let key of keys){
      if(!visited[key]) dfs(key);
    }
  }
  dfs(0);
  return visited.every(item=>item==true);
};

File: 卡码/代码随想录/Python/图论/dfs/LeetCode_0841.py
#841. 钥匙和房间
from typing import *

class Solution:
  def canVisitAllRooms(self, rooms: List[List[int]]) -> bool:
    visited:List[bool]=[False for _ in range(len(rooms))]
    def dfs(key):
      if visited[key]:
        return
      visited[key]=True
      keys:List[int]=rooms[key]
      for key in keys:
        if not visited[key]:
          dfs(key)      
    dfs(0)
    return all(visited)